src/App.tsx

```diff
@@ -116,45 +117,72 @@ const doGetCharacterData = throttle(
     setBungieServiceUnavailable: (value: boolean) => void
   ) => {
     const updateCharacterData = (newData: PowerBarsCharacterData[]) => {
-      const newOverallPower = Math.max(...newData.map(c => c.overallPower));
+      const newOverallPower = Math.max(...newData.map((c) => c.overallPower));
       const newArtifactPower = Math.max(
-        ...newData.map(c => (c.artifactData ? c.artifactData.bonusPower : 0))
+        ...newData.map((c) => (c.artifactData ? c.artifactData.bonusPower : 0))
       );
       const newTotalPower = newOverallPower + newArtifactPower;
       ga.set({
         dimension1: `${newOverallPower}`,
         dimension2: `${newArtifactPower}`,
-        dimension3: `${newTotalPower}`
+        dimension3: `${newTotalPower}`,
       });
       setCharacterData(newData);
     };
 
     const isFetchingCharacterData = getIsFetchingCharacterData();
 
     (async () => {
-      if (!isFetchingCharacterData) {
-        try {
-          await getCharacterData(updateCharacterData);
-        } catch (error) {
-          if (error instanceof BungieSystemDisabledError) {
-            setBungieSystemDisabled(true);
-          }
-          if (error.message === "503") {
-            setBungieServiceUnavailable(true);
-          }
-          console.error("Error fetching character data:", error);
+      if (isFetchingCharacterData) return;
+
+      try {
+        await getCharacterData(updateCharacterData);
+      } catch (error) {
+        if (error instanceof BungieSystemDisabledError) {
+          setBungieSystemDisabled(true);
         }
+        if (error.message === "503") {
+          setBungieServiceUnavailable(true);
+        }
+        console.error("Error fetching character data:", error);
       }
     })();
   },
   500
 );
 
+const doHardRefresh = throttle(
+  async (
+    setCharacterData: (value: PowerBarsCharacterData[]) => void,
+    setBungieSystemDisabled: (value: boolean) => void,
+    setBungieServiceUnavailable: (value: boolean) => void,
+    characterData: PowerBarsCharacterData[] | undefined
+  ) => {
+    if (characterData) {
+      try {
+        await bustProfileCache(characterData);
+      } catch (error) {
+        console.error(
+          "Error busting profile cache, but silently ignoring",
+          error
+        );
+      }
+    }
```

Comment:
Decision to just silently ignore any profile bust errors because I don't think the user needs to be aware of it. It'll just continue on to refresh the profile, maybe just a little bit more stale

(also would have liked to have done `characterData && await bustProfileCache(characterData);` but the lint config didn't like it.